libsolidity/codegen/CompilerContext.cpp

```diff
@@ -415,7 +415,7 @@ FunctionDefinition const& CompilerContext::resolveVirtualFunction(
 			if (
 				function->name() == name &&
 				!function->isConstructor() &&
-				FunctionType(*function).hasEqualParameterTypes(functionType)
+				FunctionType(*function).asCallableFunction(false)->hasEqualParameterTypes(functionType)
```

Comment:
This will match more functions than before, but as we currently do not allow mixed calldata / memory overloads, it does not change semantics.